react-components/src/components/RuleBasedOutputs/hooks/useGetDMConnectionWithNodeFromHybridMappingsQuery.test.tsx

```diff
@@ -0,0 +1,232 @@
+import { describe, it, expect, vi, beforeEach } from 'vitest';
+import { useGetDMConnectionWithNodeFromHybridMappingsQuery } from './useGetDMConnectionWithNodeFromHybridMappingsQuery';
+import { getMocksByDefaultDependencies } from '#test-utils/vitest-extensions/getMocksByDefaultDependencies';
+import { type FC, type PropsWithChildren } from 'react';
+import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
+import {
+  defaultUseGetDMConnectionWithNodeFromHybridMappingsQuery,
+  UseGetDMConnectionWithNodeFromHybridMappingsQueryContext
+} from './useGetDMConnectionWithNodeFromHybridMappingsQuery.context';
+import { type ClassicCadAssetMappingCache } from '../../CacheProvider/cad/ClassicCadAssetMappingCache';
+import { Mock } from 'moq.ts';
+import { type Node3D } from '@cognite/sdk';
+import {
+  type DmsUniqueIdentifier,
+  type FdmSDK,
+  type InspectResultList
+} from '../../../data-providers/FdmSDK';
+import { type DmCadAssetMapping } from '../../CacheProvider/cad/assetMappingTypes';
+import { type CadModelOptions } from '../../Reveal3DResources';
+import { renderHook, waitFor } from '@testing-library/react';
+import { ViewerContext } from '../../RevealCanvas/ViewerContext';
+import { type RevealRenderTarget } from '../../../architecture';
+import { type InstanceKey } from '../../../utilities/instanceIds';
+import { createCadNodeMock } from '#test-utils/fixtures/cadNode';
+
+describe(useGetDMConnectionWithNodeFromHybridMappingsQuery.name, () => {
+  const queryClient = new QueryClient();
+
+  const modelsMock: CadModelOptions[] = [
+    {
+      type: 'cad',
+      modelId: 123,
+      revisionId: 456
+    }
+  ];
+
+  const firstNode3DMock = createCadNodeMock({
+    id: 111,
+    treeIndex: 222,
+    subtreeSize: 1
+  });
+
+  const secondNode3DMock = createCadNodeMock({
+    id: 333,
+    treeIndex: 444,
+    subtreeSize: 1
+  });
+
+  const mockDmCadAssetMappingsList = createDMCadAssetMappingListMock(
+    [
+      { space: 'test-space', externalId: 'instance-1' },
+      { space: 'test-space', externalId: 'instance-2' }
+    ],
+    [firstNode3DMock, secondNode3DMock]
+  );
+
+  const mockAssetMappingDataMap = createAssetMappingDataMapMock(
+    mockDmCadAssetMappingsList.map((mapping) => mapping.instanceId),
+    [firstNode3DMock, secondNode3DMock]
+  );
+
+  const mockNoMatchedAssetMappingDataMap = createAssetMappingDataMapMock(
+    [
+      { space: 'test-space', externalId: 'other-instance-1' },
+      { space: 'test-space', externalId: 'other-instance-2' }
+    ],
+    [firstNode3DMock, secondNode3DMock]
+  );
+
+  const mockDmNodesInspectList: InspectResultList = {
+    items: [
+      {
+        space: mockDmCadAssetMappingsList[0].instanceId.space,
+        externalId: mockDmCadAssetMappingsList[0].instanceId.externalId,
```

Comment:
```suggestion
        ...mockDmCadAssetMappingsList[0].instanceId,
```
I think this would be nicer - but let me know if you disagree!